### ISSUE TITLE
Mixture of float and double parameter types

### ISSUE THREAD

**Issue by tombsar** [glfw/glfw#1125]
Currently some GLFW functions use `float` parameters, whereas others use `double`. As far as I can see all *time* and *mouse cursor* related functions use `double`, and *joystick*, *content scale*, and  *gamma* functions use `float`.

Although the extra precision is most likely irrelevant, I would be in favour of unifying these parameter types to `double` for the next major release.

Taking `glfwSetGamma` as an example, the function accepts a `float` parameter, but then promotes it to `double` implicitly.

Thoughts?

**Comment by ghuser404** [comment]
Usually libraries default to using `float`, not `double`.

**Comment by elmindreda** [comment]
Yeah, `float` unless necessary seems like a good default. Time in seconds is the only value I believe needs `double`. Cursor positions will go back to `int` in 4.0, because the only real reason for them to be `float` (XI2 high resolution positions) has already been removed.

`glfwSetGamma` does promote its argument to `double`, but only because of my old C89 habits. I'll check whether all currently supported environments provide `powf`.

**Comment by zmn28hgbn59kcmlpio8unfh7fdre523esd28q9a** [comment]
perhaps prebuilt binaries can default to float, however adding build support for double would make some users happy as well.

consider GLFWfloat or GLFWreal as a typedef or preprocessor name.
this would allow users who build their own binary to specify the typ, e.g. -DGLFWfloat double or in case of typedef they would need to manually change the type in src before compiling.
